src/Uno.UI/UI/Xaml/Window/Native/NativeWindowWrapperBase.cs

```diff
@@ -91,6 +92,38 @@ public Rect VisibleBounds
 		}
 	}
 
+	/// <summary>
+	/// The same as setting <see cref="VisibleBounds"/>, <see cref="Bounds"/> and <see cref="Size"/> but makes sure the
+	/// fired events are fired only after both properties are updated "atomically"
+	/// </summary>
+	public void SetBoundsAndVisibleBounds(Rect bounds, Rect visibleBounds)
+	{
+		var applicationView = ApplicationView.GetForWindowId(Window!.AppWindow.Id);
+		var oldBounds = _bounds;
+		var oldVisibleBounds = _visibleBounds;
+		var oldVisibleBounds2 = applicationView.VisibleBounds;
+		_bounds = bounds;
+		_visibleBounds = visibleBounds;
+		applicationView.VisibleBounds = visibleBounds;
+		if (oldBounds != bounds)
+		{
+			SizeChanged?.Invoke(this, bounds.Size);
```

Comment:
I need to review a bit deeper, during the week, but ideally we would want to avoid ApplicatioView altogether apart from notifying it about changes (AV should just source its info from VisualTree)

Reply:
I want to investigate whether the logic goes the right way here as compared to WinUI